commit e52b904b49273df35766826ce8dfb8cc3dd1ff37
Author: Dmitry Bogdanov <d.bogdanov@yadro.com>
Date:   Mon Jul 18 18:25:54 2022 +0300

    scsi: target: iscsi: Allow AuthMethod=None
    
    Allow negotiating AuthMethod=None at CSG=0 (Security Negotiation) when
    authentication is not required. That is required by the CHAP test in
    Windows HLK.
    
    Link: https://lore.kernel.org/r/20220718152555.17084-4-d.bogdanov@yadro.com
    Reviewed-by: Mike Christie <michael.christie@oracle.com>
    Signed-off-by: Dmitry Bogdanov <d.bogdanov@yadro.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/target/iscsi/iscsi_target_nego.c b/drivers/target/iscsi/iscsi_target_nego.c
index 767646438391..a167fab80588 100644
--- a/drivers/target/iscsi/iscsi_target_nego.c
+++ b/drivers/target/iscsi/iscsi_target_nego.c
@@ -814,7 +814,7 @@ static int iscsi_target_do_authentication(
 	return 0;
 }
 
-static bool iscsi_conn_auth_required(struct iscsit_conn *conn)
+bool iscsi_conn_auth_required(struct iscsit_conn *conn)
 {
 	struct iscsi_node_acl *nacl;
 	struct se_node_acl *se_nacl;